validator/src/bootstrap.rs

```diff
@@ -451,6 +452,15 @@ pub fn attempt_download_genesis_and_snapshot(
     Ok(())
 }
 
+/// simple ping helper function which returns the time to connect
+fn ping(addr: &SocketAddr) -> Option<std::time::Duration> {
+    let start = Instant::now();
+    match TcpStream::connect_timeout(addr, std::time::Duration::from_secs(2)) {
```

Comment:
Might be good to make this timeout a const (we could also use it for the RPC client timeout on 521)